include/s3select_functions.h

```diff
@@ -815,6 +817,120 @@ struct _fn_to_timestamp : public base_function
+                        "May", "June", "July", "August", "September",
+                        "October", "November", "December"};
+
+  bool operator()(bs_stmt_vec_t* args, variable* result) override
+  {
+    auto iter = args->begin();
+    int args_size = args->size();
+
+    if (args_size < 2)
+    {
+      throw base_s3select_exception("to_string need 2 parameters");
+    }
+
+    base_statement* dt1_param = *iter;
+    val_timestamp = dt1_param->eval();
+
+    if (val_timestamp.is_timestamp() == false)
+    {
+      throw base_s3select_exception("first parameter should be timestamp");
+    }
+
+    iter++;
+    base_statement* format = *iter;
+    val_format = format->eval();
+
+    if (val_format.is_string() == false)
+    {
+      throw base_s3select_exception("second parameter should be string");
+    }
+
+    boost::posix_time::ptime new_ptime = *val_timestamp.timestamp();
+    std::string fmt = val_format.to_string();
+    int64_t yr = new_ptime.date().year();
+    std::string yr_str = std::to_string(yr);
+    int64_t mnth = new_ptime.date().month();
+    std::string mnth_str = std::to_string(mnth);
+    std::string day = std::to_string(new_ptime.date().day());
+    int64_t hr = new_ptime.time_of_day().hours();
+    std::string hr_24 = std::to_string(hr);
+    std::string hr_12 = std::to_string(hr%12 == 0 ? 12 : hr%12);
+    std::string mint = std::to_string(new_ptime.time_of_day().minutes());
+    std::string sec =  std::to_string(new_ptime.time_of_day().seconds());
+
+    //std::smatch m;
+    //std::regex_match (fmt, m, yyyy);
+
+    std::string res = std::regex_replace(fmt, yyyy, yr_str);
+
+    res = std::regex_replace(res, yy, std::string(2 - std::to_string(yr%100).length(), '0') + std::to_string(yr%100));
+
+    res = std::regex_replace(res, y, yr_str);
+
+    res = std::regex_replace(res, dd, std::string(2 - day.length(), '0') + day);
+
+    res = std::regex_replace(res, d, day);
+
+    res = std::regex_replace(res, hour_h2, std::string(2 - hr_24.length(), '0') + hr_24);
+
+    res = std::regex_replace(res, hour_h1, hr_24);
+
+    res = std::regex_replace(res, hour_hh, std::string(2 - hr_12.length(), '0') + hr_12);
+
+    res = std::regex_replace(res, hour_h, hr_12);
+
+    res = std::regex_replace(res, mint_mm, std::string(2 - mint.length(), '0') + mint);
+
+    res = std::regex_replace(res, mint_m, mint);
+
+    res = std::regex_replace(res, sec_ss, std::string(2 - sec.length(), '0') + sec);
+
+    res = std::regex_replace(res, sec_s, sec);
+
+    res = std::regex_replace(res, month_mmmmm, (months[mnth - 1]).substr(0, 1));
+
+    res = std::regex_replace(res, month_mmmm, months[mnth - 1]);
+
+    res = std::regex_replace(res, month_mmm, (months[mnth - 1]).substr(0, 3));
+
+    res = std::regex_replace(res, month_mm, std::string(2 - mnth_str.length(), '0') + mnth_str);
+
+    res = std::regex_replace(res, month_m, mnth_str);
+
+    result->set_value(res.c_str());
+
```

Comment:
per each evaluation of to_string ( as the size of object, could be huge number)
it extract information from ptime ( all possible datetime parts)
and call for all possible replace commands. 

since the format is known in advance(it should be fixed), its possible to extract and replace only what is needed, 
potentially, much less.